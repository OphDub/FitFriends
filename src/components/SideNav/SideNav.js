import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../../actions/actionsIndex';
import PropTypes from 'prop-types';
import './SideNav.css';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faHome from '@fortawesome/fontawesome-free-solid/faHome';
import faUser from '@fortawesome/fontawesome-free-solid/faUser';
import faCog from '@fortawesome/fontawesome-free-solid/faCog';
import faSignOut from '@fortawesome/fontawesome-free-solid/faSignOutAlt';

export const SideNav = ({ userImage, userName, logout }) => {
  return (
    <section className="SideNav">
      <div className="user-profile">
        <img src={userImage} alt="User profile" className="user-pic"/>
        <h4 className={`${userName}`}>Brooke</h4>
      </div>
      <div className="sidenav-btns">
        <NavLink to="/home">
          <button className="sidenav-btn">
            <FontAwesomeIcon icon={faHome} size="2x"/>
            Home
          </button>
        </NavLink>
        <NavLink to="/profile">
          <button className="sidenav-btn">
            <FontAwesomeIcon icon={faUser} size="2x"/>
            Profile
          </button>
        </NavLink>
        <NavLink to="/settings">
          <button className="sidenav-btn">
            <FontAwesomeIcon icon={faCog} size="2x"/>
            Settings
          </button>
        </NavLink>
        <NavLink to="/login">
          <button className="sidenav-btn"
            onClick={logout}>
            <FontAwesomeIcon icon={faSignOut} size="2x"/>
            Logout
          </button>
        </NavLink>
      </div>
    </section>
  );
};

SideNav.propTypes = {
  userImage: PropTypes.string,
  userName: PropTypes.string,
  logout: PropTypes.func
};

export const mapStateToProps = (state) => ({
  user: state.user
});

export const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logout())
});

export default connect(mapStateToProps, mapDispatchToProps)(SideNav);