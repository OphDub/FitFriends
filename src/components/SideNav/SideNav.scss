.SideNav {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px;
  background: grey;
}

.user-pic {
  border-radius: 50%;
  width: 75px;
  height: 75px;
  padding-bottom: 5px;
}

.user-name {
  margin: 10px auto;
}

.user-profile {
  padding: 15px 0;
  display: flex;
  flex-direction: column;
}

.sidenav-btns {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.SideNav a {
  width: 100px;
  height: 30px;
  text-decoration: none;
  margin-top: 10px;
}

.sidenav-btn {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  height: 100%;
  border: none;
  background: lightgrey;
  border-radius: 10px;
}

.sidenav-btn h4 {
  margin: 0px;
}

.sidenav-btn:hover {
  color: white;
  background: #48E0EA;
}