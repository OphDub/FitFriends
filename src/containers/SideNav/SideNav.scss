.SideNav {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 10px;
  background: grey;
}

.user-pic {
  border-radius: 50%;
}

.user-name {
  margin: 10px auto;
}

.user-profile {
  padding: 15px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.sidenav-btns {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.SideNav a {
  width: 100px;
  height: 30px;
  text-decoration: none;
  margin-top: 10px;
}

.sidenav-btn {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  height: 100%;
  border: none;
  background: lightgrey;
  border-radius: 5px;
  outline: none;
}

.sidenav-btn:focus {
  outline: 0;
}

.sidenav-btn h4 {
  margin: 0px;
}

.sidenav-btn:hover {
  color: white;
  background: #48E0EA;
  cursor: pointer;
}