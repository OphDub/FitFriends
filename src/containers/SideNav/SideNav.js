import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { logout } from '../../thunks/thunks';
import PropTypes from 'prop-types';
import './SideNav.css';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faUser from '@fortawesome/fontawesome-free-solid/faUser';
import faSignOut from '@fortawesome/fontawesome-free-solid/faSignOutAlt';

export class SideNav extends Component {
  renderUserProfile = () => {
    if (this.props.user.image) {
      return (<img src={this.props.user.image} alt="User profile" className="user-pic"/>);
    }

    return (<div className="user-pic"><FontAwesomeIcon icon={faUser} size="4x"/></div>);
  }

  render () {
    return (
      <section className="SideNav">
        <div className="user-profile">
          {this.renderUserProfile()}
          <h4 className="user-name">{this.props.user.email}</h4>
        </div>
        <div className="sidenav-btns">
          <NavLink to="/login">
            <button className="sidenav-btn"
              onClick={this.props.logout}>
              <FontAwesomeIcon icon={faSignOut} size="2x"/>
              Logout
            </button>
          </NavLink>
        </div>
      </section>
    );
  }
}

const userProps = PropTypes.shape({
  name: PropTypes.string,
  image: PropTypes.string,
  email: PropTypes.string
});

SideNav.propTypes = {
  user: PropTypes.shape(userProps),
  logout: PropTypes.func
};

export const mapStateToProps = (state) => ({
  user: state.user
});

export const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(logout())
});

export default connect(mapStateToProps, mapDispatchToProps)(SideNav);