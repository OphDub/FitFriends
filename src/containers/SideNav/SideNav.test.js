/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { SideNav, mapStateToProps, mapDispatchToProps } from './SideNav';

describe('SIDENAV', () => {
  it('should match the snapshot', () => {
    const mockImgUrl = 'www.someimage.com';
    const mockUser = {
      image: mockImgUrl
    }
    const renderedComponent = shallow(<SideNav user={mockUser}/>);

    expect(renderedComponent).toMatchSnapshot();
  });

  it('should map the store correctly', () => {
    const mockUser = { loggedIn: true, name: 'Someuser@me.com'};
    const mockStore = {
      user: mockUser
    };

    const mapped = mapStateToProps(mockStore);

    expect(mapped.user).toEqual(mockStore.user);
  });

  it('should call the dispatch func when the logout func is called', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.logout();

    expect(mockDispatch).toHaveBeenCalled();
  });
});
