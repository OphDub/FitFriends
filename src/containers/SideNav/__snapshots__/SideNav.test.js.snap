// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIDENAV should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideNav
    user={
        Object {
            "image": "www.someimage.com",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-profile"
>
          <img
                    alt="User profile"
                    className="user-pic"
                    src="www.someimage.com"
          />
          <h4
                    className="user-name"
          />
</div>,
        <div
          className="sidenav-btns"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/login"
          >
                    <button
                              className="sidenav-btn"
                              onClick={undefined}
                    >
                              <FontAwesomeIcon$1
                                        border={false}
                                        className=""
                                        fixedWidth={false}
                                        flip={null}
                                        icon={
                                                  Object {
                                                            "icon": Array [
                                                              512,
                                                              512,
                                                              Array [],
                                                              "f2f5",
                                                              "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                            ],
                                                            "iconName": "sign-out-alt",
                                                            "prefix": "fas",
                                                          }
                                        }
                                        listItem={false}
                                        mask={null}
                                        name=""
                                        pull={null}
                                        pulse={false}
                                        rotation={null}
                                        size="2x"
                                        spin={false}
                                        symbol={false}
                                        transform={null}
                              />
                              Logout
                    </button>
          </NavLink>
</div>,
      ],
      "className": "SideNav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="User profile"
              className="user-pic"
              src="www.someimage.com"
/>,
            <h4
              className="user-name"
/>,
          ],
          "className": "user-profile",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "User profile",
              "className": "user-pic",
              "src": "www.someimage.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "user-name",
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/login"
>
            <button
                        className="sidenav-btn"
                        onClick={undefined}
            >
                        <FontAwesomeIcon$1
                                    border={false}
                                    className=""
                                    fixedWidth={false}
                                    flip={null}
                                    icon={
                                                Object {
                                                            "icon": Array [
                                                              512,
                                                              512,
                                                              Array [],
                                                              "f2f5",
                                                              "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                            ],
                                                            "iconName": "sign-out-alt",
                                                            "prefix": "fas",
                                                          }
                                    }
                                    listItem={false}
                                    mask={null}
                                    name=""
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size="2x"
                                    spin={false}
                                    symbol={false}
                                    transform={null}
                        />
                        Logout
            </button>
</NavLink>,
          "className": "sidenav-btns",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="sidenav-btn"
              onClick={undefined}
>
              <FontAwesomeIcon$1
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon={
                                          Object {
                                                        "icon": Array [
                                                          512,
                                                          512,
                                                          Array [],
                                                          "f2f5",
                                                          "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                        ],
                                                        "iconName": "sign-out-alt",
                                                        "prefix": "fas",
                                                      }
                            }
                            listItem={false}
                            mask={null}
                            name=""
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size="2x"
                            spin={false}
                            symbol={false}
                            transform={null}
              />
              Logout
</button>,
            "to": "/login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon$1
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                                    Object {
                                                      "icon": Array [
                                                        512,
                                                        512,
                                                        Array [],
                                                        "f2f5",
                                                        "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                      ],
                                                      "iconName": "sign-out-alt",
                                                      "prefix": "fas",
                                                    }
                  }
                  listItem={false}
                  mask={null}
                  name=""
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  transform={null}
/>,
                "Logout",
              ],
              "className": "sidenav-btn",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f2f5",
                      "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                    ],
                    "iconName": "sign-out-alt",
                    "prefix": "fas",
                  },
                  "listItem": false,
                  "mask": null,
                  "name": "",
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": "2x",
                  "spin": false,
                  "symbol": false,
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Logout",
            ],
            "type": "button",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-profile"
>
            <img
                        alt="User profile"
                        className="user-pic"
                        src="www.someimage.com"
            />
            <h4
                        className="user-name"
            />
</div>,
          <div
            className="sidenav-btns"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/login"
            >
                        <button
                                    className="sidenav-btn"
                                    onClick={undefined}
                        >
                                    <FontAwesomeIcon$1
                                                border={false}
                                                className=""
                                                fixedWidth={false}
                                                flip={null}
                                                icon={
                                                            Object {
                                                                        "icon": Array [
                                                                          512,
                                                                          512,
                                                                          Array [],
                                                                          "f2f5",
                                                                          "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                                        ],
                                                                        "iconName": "sign-out-alt",
                                                                        "prefix": "fas",
                                                                      }
                                                }
                                                listItem={false}
                                                mask={null}
                                                name=""
                                                pull={null}
                                                pulse={false}
                                                rotation={null}
                                                size="2x"
                                                spin={false}
                                                symbol={false}
                                                transform={null}
                                    />
                                    Logout
                        </button>
            </NavLink>
</div>,
        ],
        "className": "SideNav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="User profile"
                className="user-pic"
                src="www.someimage.com"
/>,
              <h4
                className="user-name"
/>,
            ],
            "className": "user-profile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "User profile",
                "className": "user-pic",
                "src": "www.someimage.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "user-name",
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/login"
>
              <button
                            className="sidenav-btn"
                            onClick={undefined}
              >
                            <FontAwesomeIcon$1
                                          border={false}
                                          className=""
                                          fixedWidth={false}
                                          flip={null}
                                          icon={
                                                        Object {
                                                                      "icon": Array [
                                                                        512,
                                                                        512,
                                                                        Array [],
                                                                        "f2f5",
                                                                        "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                                      ],
                                                                      "iconName": "sign-out-alt",
                                                                      "prefix": "fas",
                                                                    }
                                          }
                                          listItem={false}
                                          mask={null}
                                          name=""
                                          pull={null}
                                          pulse={false}
                                          rotation={null}
                                          size="2x"
                                          spin={false}
                                          symbol={false}
                                          transform={null}
                            />
                            Logout
              </button>
</NavLink>,
            "className": "sidenav-btns",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <button
                className="sidenav-btn"
                onClick={undefined}
>
                <FontAwesomeIcon$1
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon={
                                                Object {
                                                                "icon": Array [
                                                                  512,
                                                                  512,
                                                                  Array [],
                                                                  "f2f5",
                                                                  "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                                ],
                                                                "iconName": "sign-out-alt",
                                                                "prefix": "fas",
                                                              }
                                }
                                listItem={false}
                                mask={null}
                                name=""
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size="2x"
                                spin={false}
                                symbol={false}
                                transform={null}
                />
                Logout
</button>,
              "to": "/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon$1
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                                        Object {
                                                            "icon": Array [
                                                              512,
                                                              512,
                                                              Array [],
                                                              "f2f5",
                                                              "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                                                            ],
                                                            "iconName": "sign-out-alt",
                                                            "prefix": "fas",
                                                          }
                    }
                    listItem={false}
                    mask={null}
                    name=""
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size="2x"
                    spin={false}
                    symbol={false}
                    transform={null}
/>,
                  "Logout",
                ],
                "className": "sidenav-btn",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f2f5",
                        "M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z",
                      ],
                      "iconName": "sign-out-alt",
                      "prefix": "fas",
                    },
                    "listItem": false,
                    "mask": null,
                    "name": "",
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": "2x",
                    "spin": false,
                    "symbol": false,
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Logout",
              ],
              "type": "button",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
